AppendBlobReader/src/consumer.js

```diff
@@ -269,215 +249,87 @@ function getUpdatedEntity(task, endByte) {
-            context.log("Using cached accountKey for %s timeRemainingToRefreshToken: %d", task.rowKey, timeRemainingToRefreshToken);
-            resolve(cachedAccountKeyResponse[getStorageAccountCacheKey(task)]);
-        });
-    }
-}
-
-function getStorageAccountAccessKey(context, task) {
-
-    return getCachedToken(context, task).then(function(credentials) {
-        var storagecli = new storageManagementClient(
-            credentials,
-            task.subscriptionId
-        );
-        return storagecli.storageAccounts.listKeys(task.resourceGroupName, task.storageName).then(function (resp) {
-            return resp.keys[0].value;
-        });
-    });
-
-}
+    //context.log("Inside get data function:");
+    return new Promise(async function (resolve, reject) {
+        try {
+            var buffer = Buffer.alloc(task.endByte - task.startByte + 1);
+            await blobService.downloadToBuffer(buffer, task.startByte, (task.endByte - task.startByte + 1), {
+                abortSignal: AbortController.timeout(30 * 60 * 1000),
+                blockSize: 4 * 1024 * 1024,
+                concurrency: 1
+            });
+            resolve(buffer.toString());
+        } catch (err) {
+            reject(err);
+        }
+    })
+};
 
 function getBlockBlobService(context, task) {
+    return new Promise(function (resolve, reject) {
+        try {
 
-    return getCachedAccountKey(context, task).then(function (accountKey) {
-        var blobService = storage.createBlobService(task.storageName, accountKey);
-        return blobService;
-    });
-
-}
+            var containerClient = new ContainerClient(
+                `https://${task.storageName}.blob.core.windows.net/${task.containerName}`,
+                tokenCredential
+            );
+            var blockBlobClient = containerClient.getBlockBlobClient(task.blobName);
+            resolve(blockBlobClient);
+        } catch (err) {
+            reject(err);
+        }
+    })
+};
 
 function releaseLockfromOffsetTable(context, serviceBusTask, dataLenSent) {
     var curdataLenSent = dataLenSent || contentDownloaded;
     if (serviceBusTask.blobType === "AppendBlob") {
         return setAppendBlobOffset(context, serviceBusTask, curdataLenSent).catch(function (error) {
             // not failing with error because log will automatically released by appendblob
-            context.log("Error - Failed to update OffsetMap table: ", error, serviceBusTask, curdataLenSent)
+            context.log.error(`Error - Failed to update OffsetMap table, error: ${JSON.stringify(error)},  serviceBusTask: ${JSON.stringify(serviceBusTask)}, data: ${JSON.stringify(curdataLenSent)}`)
         });
     } else {
-        return new Promise(function (resolve, reject) {resolve();});
+        return new Promise(function (resolve, reject) { resolve(); });
     }
 }
 
-function releaseMessagefromDLQ(context, serviceBusTask) {
```

Comment:
why did we delete this is it not getting used?

Reply:
releaseMessagefromDLQ , its not in use any more